### PR TITLE
New Select and Reject

### LINKED ISSUE #14: Allow regular expression in Select/Reject
We should be able to write `Select(r"foo")` to select all columns containing the string "foo".

### PR DESCRIPTION
**Pull request by eliascarv**
Update Select and Reject to add Regex support.

Fix #14

### CODE REVIEW

**Comment by eliascarv** [comment]
This PR is ready for review @juliohm.

File: test/transforms.jl [hunk 1]
```diff
@@ -10,6 +10,12 @@
     f = rand(4000)
     t = Table(; a, b, c, d, e, f)
 
+    x1 = rand(4000)
+    x2 = rand(4000)
+    y1 = rand(4000)
+    y2 = rand(4000)
+    t₂ = Table(; x1, x2, y1, y2)
+
```
Comment (juliohm) [inline test/transforms.jl:18]: Let's move this to right before the corresponding test lines. We should probably stick to the surrounding convention and use t for the variable name. Every test set overwrites a new t and runs a couple of tests.

File: test/transforms.jl [hunk 2]
```diff
@@ -55,10 +61,45 @@
...
+    T = Select(["d", "c", "b"])
+    n, c = apply(T, t)
+    @test Tables.columnnames(n) == (:d, :c, :b)
+    tₒ = revert(T, n, c)
+    @test t == tₒ
+
+    # selection with Regex
+    T = Select(r"[dcb]")
+    n, c = apply(T, t)
+    @test Tables.columnnames(n) == (:b, :c, :d) # a ordem das colunas é a mesma da table original - obs: traduzir
```
Comment (juliohm) [inline test/transforms.jl:81]: Please use English always, even in comments.

File: test/transforms.jl [hunk 3]
```diff
@@ -55,10 +61,45 @@
...
+    tₒ = revert(T, n, c)
+    @test t₂ == tₒ
+
     # selection with single column
-    @test (Select(:a) == Select("a") ==
-           Select((:a,)) == Select(("a",)) ==
-           Select([:a]) == Select(["a"]))
+    @test (Select(:a).cols == Select("a").cols ==
+           Select((:a,)).cols == Select(("a",)).cols ==
+           Select([:a]).cols == Select(["a"]).cols)
```
Comment (juliohm) [inline test/transforms.jl:102]: Try to rewrite these tests without accessing internal fields. Internal fields are not part of the public API, and may change in future refactoring of the code.

**Review by juliohm — CHANGES_REQUESTED** [review]
Awesome progress. Please take a look at the attached suggestions.

Also, we should make sure that the Reject transform is also working with regex.

File: test/transforms.jl (hunk 1 above)
Comment (eliascarv) [reply test/transforms.jl]: OK, done!

File: test/transforms.jl (hunk 2 above)
Comment (eliascarv) [reply test/transforms.jl]: Done.

File: test/transforms.jl (hunk 3 above)
Comment (eliascarv) [reply test/transforms.jl]: The comparison operator has been overloaded to fix this issue.
Done!

File: src/transforms/select.jl [hunk 4]
```diff
@@ -2,29 +2,43 @@
...
 """
     Select(col₁, col₂, ..., colₙ)
     Select([col₁, col₂, ..., colₙ])
-
+    Select((col₁, col₂, ..., colₙ))
+    
 The transform that selects columns `col₁`, `col₂`, ..., `colₙ`.
+    
+    Select(Regex)
+Selects the columns that match with Regex.
```
Comment (juliohm) [inline src/transforms/select.jl:15]:
```suggestion
    Select(regex)

Selects the columns that match with `regex`.
```

File: src/transforms/select.jl [hunk 5]
```diff
@@ -2,29 +2,43 @@
...
-Select(cols...) = Select(cols)
+Select(cols::Vector{T}) where {T<:AbstractString} =
+  Select(Symbol.(cols))
+
+Base.:(==)(a::Select, b::Select) = a.cols == b.cols
 
 isrevertible(::Type{<:Select}) = true
 
+_selectedcols(cols::Vector{Symbol}, allcols) = cols
+_selectedcols(cols::Regex, allcols) =
```
Comment (juliohm) [inline src/transforms/select.jl:35]: What about renaming this function to the verb `_select`?

File: src/transforms/select.jl [hunk 6]
```diff
@@ -76,26 +90,34 @@ end
 """
     Reject(col₁, col₂, ..., colₙ)
     Reject([col₁, col₂, ..., colₙ])
+    Reject((col₁, col₂, ..., colₙ))
 
 The transform that discards columns `col₁`, `col₂`, ..., `colₙ`.
+
+    Reject(Regex)
+Discards the columns that match with Regex.
```
Comment (juliohm) [inline src/transforms/select.jl:98]:
```suggestion
    Reject(regex)

Discards the columns that match with `regex`.
```

File: src/transforms/select.jl [hunk 7]
```diff
@@ -2,29 +2,43 @@
 # Licensed under the MIT License. See LICENSE in the project root.
 # ------------------------------------------------------------------
 
+const ColSelector = Union{Vector{Symbol}, Regex}
```
Comment (juliohm) [inline src/transforms/select.jl:5]: What about renaming this to `ColSpec`? It solves the Select/Reject nomenclature issue.

**Review by juliohm — CHANGES_REQUESTED** [review]
Great PR. Just a few more changes before we can go ahead and merge.

File: src/transforms/select.jl (hunk 7 above)
Comment (eliascarv) [reply src/transforms/select.jl:5]: I think that's a good name.
I will rename `ColSelector` to `ColSpec`.

File: src/transforms/select.jl (hunk 4 above)
Comment (eliascarv) [reply src/transforms/select.jl]: Done!

File: src/transforms/select.jl (hunk 6 above)
Comment (eliascarv) [reply src/transforms/select.jl]: Done.

File: src/transforms/select.jl (hunk 7 above)
Comment (eliascarv) [reply src/transforms/select.jl]: Done!

**Review by juliohm — APPROVED** [review]

**Comment by juliohm** [comment]
Great contribution 👍🏽

**Pull request merged**